### PR TITLE
'Default' or 'case' preceded by dot (.) should not be captured as part of switch-case expression.

### PR DESCRIPTION
**Pull request by hoanhtien**
This fixes issue https://github.com/Microsoft/TypeScript-Sublime-Plugin/issues/335.

### CODE REVIEW

File: TypeScript.YAML-tmLanguage [hunk 1]
```diff
@@ -23,14 +23,14 @@ repository:
 
   control-statement:
     name: keyword.control.ts
-    match: \b(break|catch|continue|debugger|declare|do|else|finally|for|if|return|switch|throw|try|while|with|super)\b
+    match: \b(break|catch|continue|debugger|declare|do|else|finally|for|if|return|switch|throw|try|while|with|super|case|default)\b
```
Comment (billti) [inline TypeScript.YAML-tmLanguage]: I left a comment on the test-case also... but perhaps we should check keywords are not preceded by a "." . Reserved words are valid identifierNames, but not valid identifiers (i.e. "foo.return" is valid and "return" is not a keyword here, but "var return" is invalid as 'return' is a reserved work in this position).

Comment (hoanhtien) [inline TypeScript.YAML-tmLanguage]: I originally thought we wanted to keep it highlighted as keywords as Sublime does that for JS files.

**Comment by hoanhtien** [comment]
Updated the pull request to no longer colorize reserved words if they are being used as identifiers:
Before:
image
After:
image

**Comment by billti** [comment]
Thanks!

**Pull request merged**